### ISSUE TITLE
Problem when parsing expressions with dotted Identifiers and multiple BinaryExpressions

### ISSUE THREAD

**Issue by mythmon** [TechnologyAdvice/Jexl#11]
There seems to be a priority problem when multiple binary expressions are used in a row involving dotted identifiers.

```js
var jexl = require('jexl');
var context = {name: {first: 'Malory', last: 'Archer'}};
var expr = 'name.first == "Malory" && name.last == "Archer"';
jexl.eval(expr, context).catch(function(err) { console.error(err.stack); });
```

```
TypeError: Cannot read property 'type' of undefined
    at /home/mythmon/scratch/jexl-spike/node_modules/jexl/lib/evaluator/Evaluator.js:53:22
    at process._tickDomainCallback (internal/process/next_tick.js:129:7)
```

Digging around in the internals a bit, I got this AST from the parser:

```js
var Lexer = require('jexl/lib/Lexer');
var Parser = require('jexl/lib/parser/Parser');
var grammar = require('jexl/lib/grammar').elements;
var tokens = new Lexer(grammar).tokenize(expr);
var parser = new Parser(grammar);
parser.addTokens(tokens);
var ast = parser.complete();
JSON.stringify(ast, null, 2);
```

```json
{
  "type": "BinaryExpression",
  "operator": "==",
  "left": {
    "type": "Identifier",
    "value": "last",
    "from": {
      "type": "Identifier",
      "value": "name",
      "from": {
        "type": "BinaryExpression",
        "operator": "&&",
        "left": {
          "type": "BinaryExpression",
          "operator": "==",
          "left": {
            "type": "Identifier",
            "value": "first",
            "from": {
              "type": "Identifier",
              "value": "name"
            }
          },
          "right": {
            "type": "Literal",
            "value": "Malory"
          }
        }
      }
    }
  },
  "right": {
    "type": "Literal",
    "value": "Archer"
  }
}
```

That looks unbalanced to me. Specifically `left.from.left` is missing a corresponding `left.from.right`. This problem doesn't show up when using undotted identifiers.

**Comment by TomFrost** [comment]
Wow, excellent digging-- the && should definitely be the root node in this case. I'll do some exploring.

**Comment by TomFrost** [comment]
This is fixed in https://github.com/TechnologyAdvice/Jexl/commit/e82c22eae55f67a8e5fa8753c3f0dbe52152622c -- there's also a small problem with the frontend build, so I'll push a release and close this issue as soon as I'm able to correct that.

**Comment by TomFrost** [comment]
Fix published to NPM in version 1.1.3
